client-sdk/ts-web/core/test/hdkey.test.ts

```diff
@@ -0,0 +1,71 @@
+import {HDKey} from '../src/hdkey';
+
+interface TestVector {
+    seed: string;
+    tests: {
+        path?: string;
+        chainCode: string;
+        private: string;
+        public: string;
+    }[];
+}
+
+interface Adr0008Vector {
+    kind: string,
+    bip39_mnemonic: string,
+    bip_passphrase: string,
+    bip39_seed: string,
+    oasis_accounts: {
+        private_key: string,
+        public_key: string,
+        address: string
+    }[]
+}
+
+const adr0008Vectors: Adr0008Vector[] = require('./adr-0008-vectors.json');
```

Comment:
this require is going to mess up ES modules when they become mainstream. TS supports JSON `import`. Would you prefer to use that?

Reply:
It's in a test repository so it should not be published and should always run in jest's context, and using `import x from 'test.json'` makes it require an additional variable to type it, or would require manually adding declaration `.d.ts` files for those vectors which we might not want for tests only ?

Reply:
It's not about publishing. It's that if the package is published as ESM (so `type: module` in `package.json`), Node will interpret the entire package as `.js == ESM`, and hence disable `require`. If it's significant effort to get JSON imports working, I'd not worry about it, but if it's easy then it'd be a nice way to add near-future compatibility.

Reply:
Ah, I see what you mean. I'll `import` them and assign to another typed variable, it's not the prettiest but the other options are not really relevant at this time.